plugins/collections/get.js

```diff
@@ -7,42 +7,104 @@ const getSchema = schema.models.collection.get;
+ * @param   {Pipeline} pipeline  Pipeline model to get PR info for
+ * @returns {Object}
+ */
+function getPipelinePRInfo(pipeline) {
+    const prs = {
+        open: 0,
+        failing: 0
+    };
+
+    // Get all the PR jobs
+    return pipeline.getJobs({ type: 'pr' })
+        .then(prJobs => (
+            // Return array of prJobs' builds
+            Promise.all(prJobs.map(job =>
+                job.getBuilds()
+            ))
+        ))
+        .then((prJobsBuilds) => {
+            // The number of elements is the number of prs open
+            prs.open = prJobsBuilds.length;
+
+            prJobsBuilds.forEach((jobBuilds) => {
+                // Check the first element (last build) if pr job failed
+                if (jobBuilds[0].status === 'FAILURE') {
+                    prs.failing += 1;
+                }
+            });
+
+            return prs;
+        })
+        .catch(() => prs);
+}
+
+/**
+ * Helper function to get last builds of pipeline
+ *
+ * @param   {Object}  pipeline      Pipeline to get last builds for
+ * @param   {Factory} eventFactory  Factory for getting last event model
+ * @returns {Object}
+ */
+function getPipelineHealth(pipeline, eventFactory) {
+    if (!pipeline.lastEventId) {
+        return null;
+    }
+
+    return eventFactory.get(pipeline.lastEventId)
+        .then((event) => {
+            if (!event) {
+                return null;
+            }
+
+            return event.getBuilds()
+                .then((builds) => {
+                    if (builds.length) {
+                        // The events are sorted by most recent first. Need to reverse the order
+                        // to allow for matching with workflow job on the UI
+                        return builds.map(b => b.toJson()).reverse();
+                    }
+
+                    return null;
+                });
+        })
+        .catch(() => null);
+}
+
+/**
+ * Helper function to populate pipelines with last builds and pr info
+ *
+ * @param   {Array}   pipelines    Array of pipeline models to get last builds for
+ * @param   {Factory} eventFactory Factory for getting last event model
  * @returns {Array}
  */
-function getPipelinesHealth(pipelines, eventFactory) {
+function getPipelinesInfo(pipelines, eventFactory) {
     return pipelines
         // Filter out pipelines that don't exist
         .filter(pipeline => !!pipeline)
-        .map((pipeline) => {
-            const result = Object.assign({}, pipeline.toJson());
+        .map(pipeline =>
+            // Get the PR Info and last builds for each pipeline
+            Promise.all([
+                getPipelinePRInfo(pipeline),
+                getPipelineHealth(pipeline, eventFactory)
+            ])
+                // Combine the PR Info and last builds and return the populated pipeline
+                .then(([pipelinePRInfo, pipelineHealth]) => {
+                    const prInfo = {};
+                    const health = {};
```

Comment:
We don't really need 2 extra variables for this

Reply:
I was refactoring things after Jithin's comment and forgot to clean things up in this `then` block. Fixed now 👍